app/ui.js

```diff
@@ -1099,14 +1096,49 @@ var UI;
             // Don't display the connection settings until we're actually disconnected
         },
 
+        reconnect: function() {
+            delete UI.reconnect_callback;
+
+            // if reconnect has been disabled in the meantime, do nothing.
+            if (UI.inhibit_reconnect) {
+                return;
+            }
+
+            UI.connect();
+        },
+
         disconnectFinished: function (rfb, reason) {
+            var reconnect = UI.getSetting('reconnect');
+
+            if ((reconnect === 'true' || reconnect == '1') && !UI.inhibit_reconnect) {
```

Comment:
Uhm, that's not the case, there's the autoconnect function which is checked here:
```javascript
if (autoconnect === 'true' || autoconnect == '1') {
    autoconnect = true;
```

Shall I also correct that code?